torchvision/models/densenet.py

```diff
@@ -16,8 +17,17 @@
 }
 
 
+def _bn_function_factory(norm, relu, conv):
+    def bn_function(*inputs):
```

Comment:
i forgot why this concatenation is needed for checkpointing, can you remind me?

Reply:
In a densenet block, the previous outputs are concatenated with current input before passing through a layer. So the checkpoints saves memory by not saving them in the computation graph for backward pass. Instead it recomputes these intermediate activation during backward pass which makes them slower.